platforms/jvm/toolchains-jvm/src/main/java/org/gradle/jvm/toolchain/internal/install/DefaultJavaToolchainProvisioningService.java

```diff
@@ -111,17 +119,42 @@ public File tryInstall(JavaToolchainSpec spec) {
                 "Learn more about toolchain repositories at " + Documentation.userManual("toolchains", "sub:download_repositories").getUrl() + ".");
         }
 
-        for (RealizedJavaToolchainRepository request : repositories) {
-            Optional<JavaToolchainDownload> download = request.getResolver().resolve(new DefaultJavaToolchainRequest(spec, buildPlatform));
-            if (download.isPresent()) {
-                Collection<Authentication> authentications = request.getAuthentications(download.get().getUri());
-                return provisionInstallation(spec, download.get().getUri(), authentications);
+        Map<String, Exception> failedResolutions = new TreeMap<>();
+        File successfulResolution = null;
+        for (RealizedJavaToolchainRepository repository : repositories) {
+            JavaToolchainResolver resolver = repository.getResolver();
+            try {
+                Optional<JavaToolchainDownload> download = resolver.resolve(new DefaultJavaToolchainRequest(spec, buildPlatform));
```

Comment:
Only this line is the one that might be crash, right?
What I mean, this is the line to catch exceptions from toolchain provider authors.
The other four lines below are controlled by the Gradle build tool itself and therefore shouldn't be part of the try-catch. Otherwise we might add som exception message from Gradle rather than the toolchain.

Reply:
Outdated in the middle of writing the commend 😂 
Nice! 
fixed here https://github.com/gradle/gradle/pull/28205/commits/e5e0572a0521b07441945022019d35827ecdd51d

Reply:
Yes, I didn't realize that at first, hence the next set of changes.